### PR TITLE
Use TemplateNodeInfo in scale up

### PR DESCRIPTION
**Pull request by mwielgus**
ref: #43

### CODE REVIEW

File: cluster-autoscaler/core/utils.go [hunk 1]
```diff
@@ -149,37 +160,66 @@ func GetNodeInfosForGroups(nodes []*apiv1.Node, cloudProvider cloudprovider.Clou
...
-			sanitizedNodeInfo := schedulercache.NewNodeInfo(sanitizedPods...)
-			if err := sanitizedNodeInfo.SetNode(sanitizedNode); err != nil {
-				return nil, err
-			}
+		// No good template, trying to generate one. This is called only if there are no
+		// working nodes in the node groups. By default CA tries to usa a real-world example.
+		baseNodeInfo, err := nodeGroup.TemplateNodeInfo()
+		if err != nil && err != cloudprovider.ErrNotImplemented {
+			return map[string]*schedulercache.NodeInfo{}, err
+		}
```
Comment (MaciekPytel) [inline cluster-autoscaler/core/utils.go]:
```
if err == cloudprovider.ErrNotImplemented { 
    continue 
}
```
?

File: cluster-autoscaler/core/utils.go [hunk 2]
```diff
@@ -131,10 +134,18 @@ func createNodeNameToInfoMap(pods []*apiv1.Pod, nodes []*apiv1.Node) map[string]
...
-func GetNodeInfosForGroups(nodes []*apiv1.Node, cloudProvider cloudprovider.CloudProvider, kubeClient kube_client.Interface) (map[string]*schedulercache.NodeInfo, error) {
+//
+// TODO(mwielgus): Review error policy - sometimes we may continue with partial errors.
+func GetNodeInfosForGroups(nodes []*apiv1.Node, cloudProvider cloudprovider.CloudProvider, kubeClient kube_client.Interface,
+	daemonsets []*extensionsv1.DaemonSet, predicateChecker *simulator.PredicateChecker) (map[string]*schedulercache.NodeInfo, error) {
 	result := make(map[string]*schedulercache.NodeInfo)
 	for _, node := range nodes {
 
+		// Broken nodes might have some stuff missing. Skipping.
+		if !kube_util.IsNodeReadyAndSchedulable(node) {
```
Comment (MaciekPytel) [inline cluster-autoscaler/core/utils.go]: I see the point of this, but I think it has a pretty big impact. In particular I'm worried about a scenario where the only node in node group becomes unready and TemplateNodeInfo is not implemented. Wouldn't that basically be non-recoverable with this change?

File: cluster-autoscaler/core/utils.go (hunk 1 above)
Comment (MaciekPytel) [inline cluster-autoscaler/core/utils.go]: Actually can we also log before continue?

Comment (mwielgus) [inline cluster-autoscaler/core/utils.go]: Done.

File: cluster-autoscaler/core/utils.go (hunk 2 above)
Comment (mwielgus) [inline cluster-autoscaler/core/utils.go]: Removing. Next PR.

**Pull request merged**